components/modal.js

```diff
@@ -33,6 +33,7 @@ export default function useModal () {
     }
     const previousModalContent = modalStack[modalStack.length - 1]
     setModalStack(modalStack.slice(0, -1))
+    modalOptions?.onClose?.()
```

Comment:
This is required to also cancel HODL invoices in case the user clicks on `<-` and already paid:

![2024-01-16-023623_524x702_scrot](https://github.com/stackernews/stacker.news/assets/27162016/a631f30d-d6c1-4dcf-a848-7b475fd7b6fa)